src/main/java/graphql/nadel/engine/ArtificialFieldUtils.java

```diff
@@ -22,7 +23,7 @@
  */
 public class ArtificialFieldUtils {
 
-    private static final String UNDERSCORE_TYPENAME = "__typename";
+    private static final String UNDERSCORE_TYPENAME = Introspection.TypeNameMetaFieldDef.getName();
```

Comment:
I always have a philosophical dilemna with static constants.  On the one hand they should be maintainable but as a reader of code now I have to go to another place to see what this is.  Given that `__typename` is SET IN STONE by the spec I think we can leave it as is and hence its more readable even if its duplication of a value help elsewhere.

Of course another line of though its Dont Repeat YourSelf (DRY) - sooo...


Anyway I can live with this and have already spent way more words than it deserves since I am saying you dont have the change it

Reply:
I honestly would prefer to just to use  the `Introspection.TypeNameMetaFieldDef.getName()` directly